src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -269,6 +292,7 @@ private BuildEventArgs ReadTargetStartedEventArgs()
             var projectFile = ReadOptionalString();
             var targetFile = ReadOptionalString();
             var parentTarget = ReadOptionalString();
+            var buildReason = (TargetBuiltReason) ReadInt32();
```

Comment:
If we're reading an older .binlog, don't attempt to read this field.

This should be:

```
TargetBuiltReason buildReason = TargetBuiltReason.None;
if (fileFormatVersion > 3)
{
    buildReason = (TargetBuiltReason)ReadInt32();
}
```